src/libexpr/gc-small-vector.hh

```diff
@@ -2,28 +2,15 @@
 
 #include <boost/container/small_vector.hpp>
 
-#if HAVE_BOEHMGC
-
-#include <gc/gc.h>
-#include <gc/gc_cpp.h>
-#include <gc/gc_allocator.h>
-
-#endif
+#include "value.hh"
```

Comment:
I'd like for this header to be more of a "leaf" considering its narrow responsibility, so importing `value.hh` here is not great. How about we put it in `eval-gc.hh`?